Automattic/jetpack
# Monorepo: add VaultPress plugin

#### Changes proposed in this Pull Request:

Instead of working on the VaultPress plugin in its own repo, let's maintain it here, in the monorepo.

This PR brings everything from https://github.com/Automattic/vaultpress/
From now on, changes made here in the monorepo should get pushed to that same repo, that will become the production repo.

This PR:
- adds the plugin base.
- adds most of the plugin files to the phpcs exclusion list.
- adds the plugin files.
- updates the plugin's changelog to match the new format.
- updates the plugin's dependencies to match the format we use here in the monorepo.
- adds the necessary attributes so we can release the plugin from here.
- adds a changelog entry.

#### Jetpack product discussion

* Internal reference: pIFnH-6EL-p2

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

* In your local Docker environment, build the VaultPress plugin: `jetpack build plugins/vaultpress`
* When going to the Plugins menu, you should now be able to activate the VaultPress plugin.
* Once activated, you should see a VaultPress submenu under the Jetpack menu.

* Do the checks below pass?
